File: sdk/clojure/src/rocksky/album.clj
(ns rocksky.album
  "Endpoints under app.rocksky.album.*"
  (:require [rocksky.client :as c]))

(defn get-album
  "Get detailed album view.

  Required: `:uri` (AT-URI)."
  [client {:keys [uri]}]
  (c/query client :app.rocksky.album.getAlbum {:uri uri}))

(defn get-albums
  "Get albums.

  Optional: `:limit` `:offset` `:genre`."
  [client {:keys [limit offset genre]}]
  (c/query client :app.rocksky.album.getAlbums
           {:limit limit :offset offset :genre genre}))

(defn get-album-tracks
  "Get tracks for an album.

  Required: `:uri`."
  [client {:keys [uri]}]
  (c/query client :app.rocksky.album.getAlbumTracks {:uri uri}))

File: sdk/elixir/lib/rocksky/album.ex
defmodule Rocksky.Album do
  @moduledoc "`app.rocksky.album.*` endpoints."

  alias Rocksky.HTTP

  @doc "Fetch an album by AT-URI (`:uri`)."
  def get_album(client, params \\ []),
    do: HTTP.query(client, "app.rocksky.album.getAlbum", params)

  @doc "Paginated list of albums. Params: `:limit`, `:offset`, `:genre`."
  def get_albums(client, params \\ []),
    do: HTTP.query(client, "app.rocksky.album.getAlbums", params)

  @doc "Tracks belonging to an album. Params: `:uri`."
  def get_album_tracks(client, params \\ []),
    do: HTTP.query(client, "app.rocksky.album.getAlbumTracks", params)
end
